### PR TITLE
feat(shorebird_cli): upgrade to Flutter 3.10.4

### PR DESCRIPTION
**Pull request by felangel**
## Description

- feat(shorebird_cli): upgrade to Flutter 3.10.4

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by bryanoltman — APPROVED** [review]

**Comment by felangel** [comment]
Going to build/upload artifacts before merging

**Comment by felangel** [comment]
Won't merge until we've revved the revision to include iOS cleanup and built/uploaded artifacts

**Review by eseidel — APPROVED** [review]

File: bin/internal/flutter.version
Comment (bryanoltman) [inline bin/internal/flutter.version]: Is this correct? Looking at https://github.com/shorebirdtech/flutter/tree/flutter_release/3.10.4, HEAD is `cf97c772eecef5cdc388f5457631fb4ae071e68d`

**Review by bryanoltman — APPROVED** [review]

**Pull request merged**
